csharp/src/Ice/Communicator-Properties.cs

```diff
@@ -53,8 +53,16 @@ internal PropertyValue(string v, bool u)
             {
                 if (_properties.TryGetValue(name, out PropertyValue? pv))
                 {
-                    pv.Used = true;
-                    return int.Parse(pv.Val, CultureInfo.InvariantCulture);
+                    try
+                    {
+                        pv.Used = true;
+                        return int.Parse(pv.Val, CultureInfo.InvariantCulture);
+                    }
+                    catch (Exception ex) when(ex is FormatException || ex is OverflowException)
```

Comment:
I think the `when` is unnecessary, since Parse cannot throw anything else.

Reply:
What do you mean? It can throw both a FormatException and an OverflowException.
Here, exception filtering with a `when` seems better than having 2 separate catches to me.

Reply:
I mean you don't need any when. Just catch Exception and throw InvalidConfigurationException.

Reply:
But what about accidentally catching a critical exception that has nothing to do with parsing, say a StackOverflowException or something? Or do you think this to be over-zealous and pedantic?

Reply:
If it's a critical exception, we'll die soon enough. And here we're just wrapping the exception thrown by int.Parse, not discarding it.

Reply:
Alright, I'll remove the filtering, even if it hurts me a little.